Integrations/AutofocusV2/AutofocusV2.py

```diff
@@ -0,0 +1,713 @@
+                "field": "sample.tag_class",
+                "operator": "is",
+                "value": tag_class
+            }
+        ]
+    }
+    tag_scopes = list()
+    if private:
+        tag_scopes.append('private')
+    if public:
+        tag_scopes.append('public')
+    if commodity:
+        tag_scopes.append('commodity')
+    if unit42:
+        tag_scopes.append('unit42')
+    data = {
+        'query': query,
+        'scope': scope,
+        'tagScopes': tag_scopes
+    }
+    path = '/top-tags/search/'
+    resp = http_request(path, data=data, err_operation='Top tags operation failed')
+    in_progress = resp.get('af_in_progress')
+    status = 'in progress' if in_progress else 'complete'
+    search_info = {
+        'AFCookie': resp.get('af_cookie'),
+        'Status': status
+    }
+    return search_info
+
+
+def parse_top_tags_response(response):
+    top_tags_list = []
+    for tag in response.get('top_tags'):
+        fields_to_extract_from_top_tags = ['tag_name', 'public_tag_name', 'count', 'lasthit']
+        new_tag = {}
+        for field in fields_to_extract_from_top_tags:
+            new_tag[field] = tag[field]
+        top_tags_list.append(new_tag)
+    return top_tags_list
+
+
+def get_top_tags_results(af_cookie):
+    path = f'/top-tags/results/{af_cookie}'
+    results = http_request(path, err_operation='Fetching top tags results failed')
+    top_tags = parse_top_tags_response(results)
+    in_progress = results.get('af_in_progress')
+    status = 'in progress' if in_progress else 'complete'
+    return top_tags, status
+
+
+def print_hr_by_category(category_name, category_data):
+    hr = content = f'### {string_to_table_header(category_name)}:\nNo entries'
+    if category_name == 'coverage':
+        content = category_data
+        if category_data:
+            hr = tableToMarkdown(f'{string_to_table_header(category_name)}:', category_data,
+                                 headerTransform=string_to_table_header)
+        else:
+            hr = f'### {string_to_table_header(category_name)}:\nNo entries'
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['text'],
+            'Contents': content,
+            'HumanReadable': hr
+        })
+    else:
+        for os_name, os_data in category_data.items():
+            content = os_data
+            table_header = f'{category_name}_{os_name}'
+            if os_data:
+                hr = tableToMarkdown(f'{string_to_table_header(table_header)}:', os_data,
+                                     headerTransform=string_to_table_header)
+            else:
+                hr = f'### {string_to_table_header(table_header)}:\nNo entries'
+            demisto.results({
+                'Type': entryTypes['note'],
+                'ContentsFormat': formats['text'],
+                'Contents': content,
+                'HumanReadable': hr
+            })
+
+
+''' COMMANDS'''
+
+
+def test_module():
```

Comment:
need to check that wrong host and bad auth are creating readable errors

Reply:
There is no host parameter at all and the API key is already being checked. 
The integration is in Demo2 you can play with the test module.